packages/notifier/test/test-notifier.js

```diff
@@ -0,0 +1,74 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from 'tape-promise/tape';
+import { makeNotifier } from '../src/notifier';
+
+test('notifier - initital state', t => {
+  const notifier = makeNotifier();
+  notifier.updateState(1);
+
+  const updateDeNovo = notifier.getUpdateSince();
+  const updateFromNonExistent = notifier.getUpdateSince({});
+
+  t.equals(updateDeNovo.currentState, 1, 'state is one');
+  t.deepEquals(updateDeNovo, updateFromNonExistent, 'no param same as unknown');
+  t.end();
+});
+
+test('notifier - single update', t => {
+  t.plan(3);
+  const notifier = makeNotifier();
+  notifier.updateState(1);
+
+  const updateDeNovo = notifier.getUpdateSince();
+
+  const updateInWaiting = notifier.getUpdateSince(updateDeNovo.updateHandle);
+  t.equals(updateDeNovo.currentState, 1, 'initial state is one');
+  Promise.all([updateInWaiting]).then(([update]) => {
+    t.equals(update.currentState, 3, 'updated state is eventually three');
+  });
+
+  t.equals(notifier.getUpdateSince({}).currentState, 1);
+  notifier.updateState(3);
+});
+
+test('notifier - update after state change', t => {
+  t.plan(5);
+  const notifier = makeNotifier();
+  notifier.updateState(1);
+
+  const updateDeNovo = notifier.getUpdateSince();
+
+  const updateInWaiting = notifier.getUpdateSince(updateDeNovo.updateHandle);
+  t.equals(updateDeNovo.currentState, 1, 'first state check (1)');
+  Promise.all([updateInWaiting]).then(([update1]) => {
+    t.equals(update1.currentState, 3, '4th check (delayed) 3');
+    const thirdStatePromise = update1.nextStatePromise;
+    Promise.all([thirdStatePromise]).then(([update2]) => {
+      t.equals(update2.currentState, 5, '5th check (delayed) 5');
+    });
+  });
+
+  t.equals(notifier.getUpdateSince({}).currentState, 1, '2nd check (1)');
+  notifier.updateState(3);
+
+  t.equals(notifier.getUpdateSince({}).currentState, 3, '3rd check (3)');
+  notifier.updateState(5);
+});
+
+test('notifier - final state', t => {
+  t.plan(5);
+  const notifier = makeNotifier();
+  notifier.updateState(1);
+
+  const updateDeNovo = notifier.getUpdateSince();
+  const updateInWaiting = notifier.getUpdateSince(updateDeNovo.updateHandle);
+  t.equals(updateDeNovo.currentState, 1, 'initial state is one');
+  Promise.all([updateInWaiting]).then(([update]) => {
+    t.equals(update.currentState, 'final', 'state is "final"');
+    t.notOk(update.nextStatePromise, 'no next promise after final');
```

Comment:
In more async scenarios, where this record appears before the stream is closed, this will be a promise that eventually fulfills to undefined.

Reply:
I think I'm wrong about closing the stream and your code is correct.